libraries/core/lib/search/objective/BranchObjectiveFunction.ts

```diff
@@ -66,95 +60,41 @@ export class BranchObjectiveFunction<T extends Encoding>
       return Number.MAX_VALUE;
     }
 
-    // let's check if the line is covered
-    if (executionResult.coversLine(this._line)) {
-      const branchTrace = executionResult
-        .getTraces()
-        .find(
-          (trace) =>
-            trace.type === "branch" &&
-            trace.line === this._line &&
-            trace.branchType === this._type
-        );
-
-      if (branchTrace.hits > 0) {
-        return 0;
-      } else {
-        const oppositeBranch = executionResult.getTraces().find(
-          (trace) =>
-            trace.type === "branch" &&
-            trace.id === branchTrace.id && // Same branch id
-            trace.branchType !== this._type // The opposite branch type
-        );
-
-        return BranchDistance.branchDistanceNumeric(
-          oppositeBranch.opcode,
-          oppositeBranch.left,
-          oppositeBranch.right,
-          this._type
-        );
-      }
+    // let's check if the node is covered
+    if (executionResult.coversId(this._id)) {
+      return 0;
     }
 
-    // find the corresponding branch node inside the cfg
-    const branchNode = this._subject.cfg.findNodeOfTypeByLine(
-      this._line,
-      NodeType.Branch
-    );
+    // find the corresponding node inside the cfg
+    const targetNode = this._subject.cfg.graph.getNodeById(this._id);
```

Comment:
This would still target the `childNode`, right?

Reply:
Yes, but if the direct parent is covered the approach level is 0 so it should work.